client/dashboard/app/hooks/site-performance.ts

```diff
@@ -1,64 +1,93 @@
-export function usePerformanceData( siteId: number, url: string ): PerformanceData {
-	const { data: siteSettings, isLoading: isLoadingSiteSettings } = useQuery( {
-		...siteSettingsQuery( siteId ),
-		refetchOnWindowFocus: false,
-		retry: false,
-		enabled: !! siteId,
-	} );
+/**
+ * Checks if the report is failed.
+ * @param report - The report to check.
+ * @returns True if the report is failed, false otherwise.
+ */
+const isReportFailed = ( report: unknown ) => report === 'failed';
 
-	const wpcomPerformanceReportUrl: string = siteSettings?.wpcom_performance_report_url || '';
-	const [ , cachedHash ] = wpcomPerformanceReportUrl.split( '&hash=' );
+/**
+ * Checks if the report is running. We consider a report running if it is 'running' or 'queued'.
+ * @param report - The report to check.
+ * @returns True if the report is running, false otherwise.
+ */
+const isReportRunning = ( report: unknown ) => 'running' === report || 'queued' === report;
 
-	const { data: basicMetricsData, isLoading: isLoadingBasicMetrics } = useQuery( {
-		...basicMetricsQuery( url ),
+export function usePerformanceData(
+	url: string | undefined,
+	hash: string | undefined
+): PerformanceData {
+	const {
+		data: basicMetricsData,
+		isLoading: isLoadingBasicMetrics,
+		isError: isBasicMetricsError,
+		refetch,
+	} = useQuery( {
+		...basicMetricsQuery( url as string ),
 		refetchOnWindowFocus: false,
-		enabled: !! url && ! isLoadingSiteSettings && ! cachedHash,
+		enabled: !! url && ! hash,
 	} );
 
-	const token = cachedHash || basicMetricsData?.token;
+	const token = basicMetricsData?.token || hash;
 
-	const { data: performanceData, isLoading: isLoadingPerformanceInsights } = useQuery( {
-		...performanceInsightsQuery( url, token || '' ),
+	const {
+		data: performanceData,
+		isLoading: isLoadingPerformanceInsights,
+		isError: isInsightsError,
+	} = useQuery( {
+		...performanceInsightsQuery( url as string, token || '' ),
 		refetchOnWindowFocus: false,
+		retry: false,
 		enabled: !! url && !! token,
 	} );
 
 	const desktopLoaded = typeof performanceData?.pagespeed?.desktop === 'object';
 	const mobileLoaded = typeof performanceData?.pagespeed?.mobile === 'object';
 
-	const desktopScore =
-		desktopLoaded && typeof performanceData?.pagespeed.desktop === 'object'
-			? Math.round( performanceData.pagespeed.desktop.overall_score * 100 )
+	const desktopReport =
+		typeof performanceData?.pagespeed?.desktop === 'object'
+			? performanceData.pagespeed.desktop
 			: undefined;
 
-	const mobileScore =
-		mobileLoaded && typeof performanceData?.pagespeed.mobile === 'object'
-			? Math.round( performanceData.pagespeed.mobile.overall_score * 100 )
+	const mobileReport =
+		typeof performanceData?.pagespeed?.mobile === 'object'
+			? performanceData.pagespeed.mobile
 			: undefined;
 
 	return {
-		performanceData,
-		desktopScore,
-		mobileScore,
+		hash: token,
+		mobileReport,
+		desktopReport,
+		desktopScore: performanceData?.pagespeed.desktop?.overall_score,
```

Comment:
The rest of the code has optional chaining for pagespeed so it's probably needed here as well.
```suggestion
		desktopScore: performanceData?.pagespeed?.desktop?.overall_score,
```